utils/python/CIME/XML/generic_xml.py

```diff
@@ -230,11 +240,9 @@ def get_resolved_value(self, raw_value):
             elif var == "SRCROOT":
                 srcroot = os.path.join(get_cime_root(),"..")
                 item_data = item_data.replace(m.group(), srcroot)
-            elif var in os.environ:
```

Comment:
Significant change, please look at closely. I didn't like how we silently fell back to using the environment even if the user did not do the correct $ENV syntax.